src/retrace/retrace.py

```diff
@@ -295,63 +385,54 @@ def run_gdb(savedir: Union[str, Path], plugin, repopath: str, taskid: int):
         child = run(["/usr/bin/mock", "chroot", "--configdir", savedir,
                      "--", "su mockbuild -c '/bin/sh /var/spool/abrt/gdb.sh' 2>&1"],
                     # redirect GDB's stderr, ignore mock's stderr
-                    stdout=PIPE, stderr=DEVNULL, encoding='utf-8', check=False)
+                    stdout=PIPE, stderr=DEVNULL, encoding="utf-8", check=False)
 
         if child.returncode:
             raise Exception("Running GDB failed")
 
         backtrace = child.stdout.strip()
 
     elif CONFIG["RetraceEnvironment"] == "podman":
-        podman_build_call = [PODMAN_BIN, "build",
-                             "--quiet",
-                             "--force-rm",
-                             "--file", str(savedir / RetraceTask.DOCKERFILE),
-                             "--volume=%s:%s:ro" % (repopath, repopath)]
+        from .backends.podman import LocalPodmanBackend
 
-        if CONFIG["RequireGPGCheck"]:
-            podman_build_call.append("--volume=%s:%s:ro" % (RETRACE_GPG_KEYS, RETRACE_GPG_KEYS))
+        backend = LocalPodmanBackend(retrace_config=CONFIG)
 
-        if CONFIG["UseFafPackages"]:
-            faf_link_dir = CONFIG["FafLinkDir"]
-            log_debug("Using FAF repository")
-            podman_build_call.append("--volume=%s:%s" % (faf_link_dir, faf_link_dir))
+        # Start container from prepared release-specific image.
+        with backend.start_container(image_tag, taskid, repopath) as container:
+            # Copy coredump from crash directory to container.
```

Comment:
This is nothing new, we used to copy the coredump in the previous implementation as well. Since the container is ephemeral, it shouldn't pose any long-term issues when it comes to storage, and it should be reasonably fast even for larger coredumps.

If need be, I suppose we could mount the directory where the coredump resides as a volume and avoid the copy. However, that seems to me to add unnecessary complexity at the moment.